presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -148,7 +148,7 @@ public static PlanNodeStatsEstimate intersect(PlanNodeStatsEstimate left, PlanNo
                             .setNullsFraction(min(leftSymbolStats.getNullsFraction(), rightSymbolStats.getNullsFraction()))
                             .build());
         }
-
+        statsBuilder.setOutputRowCount(left.getOutputRowCount() + right.getOutputRowCount());  // this is the maximum row count;
```

Comment:
I mean that it seems like this should be separate commit. As it just fixes the issue of missing output row count estimate for INTERSECT (btw - test would be nice for that). And the commit in this PR says about adding capping of output rowCount to input rowCount.